apps/rebar/src/rebar_compiler_dag.erl

```diff
@@ -149,12 +149,21 @@ finalise_populate_sources_(G, InDirs, [{Status, {deps, Source, AbsIncls}}|Acc])
                                   {_, _Src, Path, _Label} <- [digraph:edge(G, Edge)],
                                   not lists:member(Path, AbsIncls)],
     %% Add the rest
-    [digraph:add_edge(G, Source, Incl) || Incl <- AbsIncls],
+    RemainingEdges = [digraph:edge(G, E) || E <- digraph:out_edges(G, Source),
+                                            AbsIncls =/= []],
+    [digraph:add_edge(G, Source, Incl) || Incl <- AbsIncls,
+                                          not in_edges(Source, Incl, RemainingEdges)],
```

Comment:
So add it if its not already in the edges?

Reply:
yeah. Problem was that edges can be duplicated and I assumed they weren't.

Reply:
Ok, surprised me too seeing this patch.